tests/service/test_service.yml

```diff
@@ -0,0 +1,544 @@
+# This test uses skip_host_check, so it will fail if not using
+# FreeIPA version 4.7.0 or later.
+#
+# To test against earlier versions, use test_without_skip_host_check.yml.
+#
+---
+- name: Test service
+  hosts: ipaserver
+  become: yes
+
+  tasks:
+  # setup
+  - name: Get Domain from server name
+    set_fact:
+      ipaserver_domain: "{{ groups.ipaserver[0].split('.')[1:] | join ('.') }}"
+    when: ipaserver_domain is not defined
+
+  - name: Set host1, host2 and svc hosts fqdn
+    set_fact:
+      host1_fqdn: "{{ 'host1.' + ipaserver_domain }}"
+      host2_fqdn: "{{ 'host2.' + ipaserver_domain }}"
+      svc_fqdn: "{{ 'svc.' + ipaserver_domain }}"
+
+  - name: Host absent
+    ipahost:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - www.ansible.com
+      - no.idontexist.info
+      - svc.ihavenodns.info
+      - "{{ host1_fqdn }}"
+      - "{{ host2_fqdn }}"
+      - "{{ svc_fqdn }}"
+      update_dns: no
+      state: absent
+
+  # If ansible-freeipa dnsrecord module is availble, it can be used.
+  - name: DNS record is not present.
+    command: ipa dnsrecord-del --del-all "{{ ipaserver_domain }}" svc
+    ignore_errors: yes
+
+  - name: Get IPv4 address prefix from server node
+    set_fact:
+      ipv4_prefix: "{{ ansible_default_ipv4.address.split('.')[:-1] |
+                       join('.') }}"
+
+  # If ansible-freeipa dnsrecord module is availble, it can be used.
+  - name: DNS record is present.
+    command: ipa dnsrecord-add --a-ip-address="{{ ipv4_prefix + '.201' }}" "{{ ipaserver_domain }}" svc
+
+  - name: Host "{{ host1_fqdn }}" present
+    ipahost:
+      ipaadmin_password: SomeADMINpassword
+      name: "{{ host1_fqdn }}"
+      force: yes
+
+  - name: Host "{{ host2_fqdn }}" present
+    ipahost:
+      ipaadmin_password: SomeADMINpassword
+      name: "{{ host2_fqdn }}"
+      force: yes
+
+  - name: Host "{{ svc_fqdn }}" present
+    ipahost:
+      ipaadmin_password: SomeADMINpassword
+      name: "{{ svc_fqdn }}"
+
+  - name: Host svc.ihavenodns.info present
+    ipahost:
+      ipaadmin_password: SomeADMINpassword
+      name: svc.ihavenodns.info
+      force: yes
```

Comment:
You could also add all the hosts in a single task using `hosts` dict.

Reply:
Done.